### DISCUSSION TITLE
Multiple checkboxes using array of ID's not behaving as expected

### DISCUSSION THREAD

**Discussion by zackperdue** [react-hook-form/react-hook-form#3836] [Q&A]
I'm building a tag feature where in the form I'll display a list of checkboxes which correspond to different tags.

The record who's tags are updated requires the tag data to be submitted in the shape of:

```
const payload = { tag_ids: [3, 5, 7, 9] }
```

Right now the form looks like this:
```
{tags.map(({ id, name }) => (
    <input type="checkbox" name="tag_ids[]" value={id} ref={register} />
))
```

This works well for choosing and submitting the data... it works as expected.

However when I refresh, even though the form defaultValues is supplied with `tag_ids: [3, 5, 7, 9]` the checkboxes aren't checked like they should be.

Any ideas?

**Comment by bluebill1049** [comment]
- try without `[]`, just `tag_ids`
- build a controlled input and wrapped with Controller

**Reply by zackperdue** [reply]
Ok, after trying `Controller`, `useForm` seems to have the correct `tag_ids` internally but each Input is handed the entire array of ids and not it's specific value.

Not sure what I should do next.

**Comment by zackperdue** [comment]
Just thought I'd throw this in here... @bluebill1049 let me know if this is a decent solution.

```
<Controller render={(
    { value, onChange }
) => {

    return tags.map(({ id, name }) => {
        return (
            <div key={id}>
                <label>
                    <input
                        type="checkbox"
                        value={id}
                        onChange={(e) => {
                            onChange(e.target.checked ? union(value, [e.target.value | 0]) : difference(value, [e.target.value | 0]))
                        }}
                        checked={includes(value, id)}
                        className="mr-2" />
                    {name}
                </label>
            </div>
        )
    })

}}
    control={control}
    name="tag_ids"
/>
```

Id's are integers. I used `union` & `difference` from lodash to create a unique array and update the value using `onChange` depending on if it was checked or not.
